packages/server/test/headers.test.tsx

```diff
@@ -13,9 +13,13 @@ test('pass headers', async () => {
   const { close, httpUrl } = routerToServerAndClient(
     trpc.router<Context>().query('hello', {
       resolve({ ctx }) {
-        return {
-          'x-special': ctx.headers['x-special'],
-        };
+        // return only custom headers
+        return Object.keys(ctx.headers).reduce((acc, key) => {
+          if (key.startsWith('x-')) {
+            return { ...acc, [key]: ctx.headers[key] };
+          }
+          return acc;
+        }, {});
```

Comment:
This is because previously `query('hello', ...)` was only responding with `{ 'x-special': string | undefined }`, however the new test I have added below would like to check the `'x-defined'` & `'x-undefined'` headers. I have therefore added this block to return all custom headers back to client (only if they start with 'x-'). Hope this is ok?

Reply:
Could change to this if you prefer?

```
return {
  'x-special': ctx.headers['x-special'],
  'x-defined': ctx.headers['x-defined'],
  'x-undefined': ctx.headers['x-undefined'],
};
```